DEVELOPMENT.md

```diff
@@ -0,0 +1,42 @@
+# Setting up a development environment
+
+For setting up a development environment you will need two things:
+
+- Kubernetes workload clusters
+- A kcp cluster
+
+We have therefore two scripts to get them running on your local machine with minimal resource consumption.
+
+Prerequisites:
+- [kind](https://github.com/kubernetes-sigs/kind)
+- a container engine: [podman](https://podman.io/) or [docker](https://docs.docker.com/engine/)
+- [git](https://git-scm.com/)
+
+## Workload clusters with kind
+
+[./local/kind/setup.sh](./local/kind/setup.sh) will create two kind clusters and the files to register them so that it is straightforward to add them to the kcp cluster. Kind will also output the kubeconfig files that can be used to interact with the Kubernetes clusters. Simply run:
+
+```console
+./local/kind/setup.sh
```

Comment:
> Is there any step I am missing?

What was the output of the script?

> If you run the script more than once, it is creating new folders under temp each time. So maybe, we could check if it exists or delete old ones once we are done using them..?

The script checks whether the kind clusters already exist and does not re-provision them in that case. I am not for deleting things without the user doing it on purpose to avoid bad surprise.

> Is there any reason to choose /tmp folder to create the config files of kind cluster? So far in the repo, we just create a folder called 'work' and place everything there. And work folder is not committed to git.

Yes the data is short lived and as such belongs to /tmp. This makes it easy to administer the system and avoid scattered data eating the storage capacity. Also if ever some delete operation is to be done risks are limited under /tmp. Deleting something under "work" can have very annoying consequences on the user machine... if it happens that "work" is the name of another directory there and not the one expected by the script. I have had first hand experience with that.

Reply:
Thanks @fgiloux for fixing this. It now works locally with support for docker and podman.